src/Common/src/Interop/Unix/System.Private.CoreLib.Native/Interop.Memory.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal unsafe partial class Sys
+    {
+        [DllImport(Interop.Libraries.CoreLibNative, EntryPoint = "CoreLibNative_MemAlloc")]
```

Comment:
I think we should have separate files:

- Interop.MemAllocFree.cs - that has malloc and free
- Interop.MemRealloc.cs - that has realloc